### PR TITLE
WIP: irmin-bench: Use well defined traces

### PR DESCRIPTION
**Pull request by Ngoguey42**
My progress on that matter. Comments are welcome.

### CODE REVIEW

File: bench/irmin-pack/trace_collection.ml [hunk 1]
```diff
@@ -10,7 +10,7 @@ let ( >|= ) = Lwt.Infix.( >|= )
 let ( let* ) = ( >>= )
 let ( let+ ) = ( >|= )
 
-module type STORE = Irmin.S with type key = string list
+module type STORE = Irmin.S with type key = string list and type step = string
```
Comment (samoht) [inline bench/irmin-pack/trace_collection.ml:13]: You probably want `Irmin.KV` here?

**Comment by samoht** [comment]
Can you add file headers?

Also why did you put this in `irmin-pack`? That seems like a pretty generic trace format. Could we use it for recording any running Irmin instance? Maybe move this in `irmin-trace` if that's the case?

**Comment by Ngoguey42** [comment]
Generic to any "trace" use case. We could use those to record anything.
```
trace_stat_summary_utils.ml
trace_common.ml
```

Specific to `trace_stat`, following a bootstrap trace replay or a tezos-node play.
```
trace_stat_summary.ml
trace_stat_summary_conf.ml
trace_stat_summary_pp.ml
trace_stats.ml (a bin)
```

Specific to both `trace_stat` and `trace_replayable`, but could be split.
```
trace_collection.ml
trace_definitions.ml
```


I like the idea of moving this to a `irmin-trace`, but is it worth it?

**Comment by samoht** [comment]
> I like the idea of moving this to a irmin-trace, but is it worth it?

I think it's worth it in the long-term, as it would allow anyone running the irmin CLI to enable tracing for any kind of backends. But it's clearly not high-priority, so this should not block the merge of this PR :-)

**Comment by Ngoguey42** [comment]
> I think it's worth it in the long-term, as it would allow anyone running the irmin CLI to enable tracing for any kind of backends.

I'll keep this in mind

**Comment by samoht** [comment]
(or in an issue :p)

**Pull request closed without merging**